nova_vm/src/ecmascript/execution/execution_context.rs

```diff
@@ -93,6 +96,36 @@ impl ExecutionContext {
     }
 }
 
+impl HeapMarkAndSweep for ECMAScriptCodeEvaluationState {
+    fn mark_values(&self, queues: &mut WorkQueues) {
+        self.lexical_environment.mark_values(queues);
+        self.variable_environment.mark_values(queues);
+        self.private_environment.mark_values(queues);
+    }
+
+    fn sweep_values(&mut self, compactions: &CompactionLists) {
+        self.lexical_environment.sweep_values(compactions);
+        self.variable_environment.sweep_values(compactions);
+        self.private_environment.sweep_values(compactions);
+    }
+}
+
+impl HeapMarkAndSweep for ExecutionContext {
+    fn mark_values(&self, queues: &mut WorkQueues) {
+        self.ecmascript_code.mark_values(queues);
+        self.function.mark_values(queues);
+        self.realm.mark_values(queues);
+        self.script_or_module.mark_values(queues);
```

Comment:
**question:** Is this missing the source code marking? IIRC I needed to add that to this struct.

Reply:
The implementation for `ECMAScriptCodeEvaluationState` is missing it indeed. I think this was an issue with rebasing.